lib/Dialect/Torch/Transforms/DecomposeComplexOps.cpp

```diff
@@ -37,32 +37,34 @@ static bool isNoneOrFloatDtype(MLIRContext *context, Value dtype) {
 }
 
 // Helper function to compute the return type of the reduction function.
-// `dim` specifies the dimension to reduce and `keepDim` preserves the rank of
-// the input tensor.
+// `dims` specifies the dimensions to be reduced and `keepDim` preserves the
+// rank of the input tensor.
 static Type computeReductionType(PatternRewriter &rewriter, Operation *op,
-                                 Value input, Value dim, bool keepDim) {
+                                 Value input, ValueRange dims, bool keepDim) {
   BaseTensorType tensorType = input.getType().cast<BaseTensorType>();
   SmallVector<int64_t> sizes;
   int64_t dimInt;
   if (tensorType.hasSizes()) {
     ArrayRef<int64_t> inputShape = tensorType.getSizes();
     int64_t inputRank = inputShape.size();
-    if (matchPattern(dim, m_TorchConstantInt(&dimInt))) {
-      dimInt = toPositiveDim(dimInt, inputRank);
-      if (!isValidDim(dimInt, inputRank)) {
-        (void)rewriter.notifyMatchFailure(op, "dim is not a valid dim");
-        return nullptr;
+    unsigned reducedRank = keepDim ? inputRank : inputRank - dims.size();
+    sizes.append(inputShape.begin(), inputShape.end());
+    for (auto dim : dims) {
+      if (matchPattern(dim, m_TorchConstantInt(&dimInt))) {
+        dimInt = toPositiveDim(dimInt, inputRank);
+        if (!isValidDim(dimInt, inputRank)) {
+          (void)rewriter.notifyMatchFailure(op, "dim is not a valid dim");
+          return nullptr;
+        }
+        // The dimension to be reduced is set to 1 when `keepDim` is true else
+        // it is removed.
+        if (keepDim)
+          sizes[dimInt] = 1;
+        else
+          sizes.erase(sizes.begin() + dimInt - 1);
```

Comment:
I don't think this is correct. Suppose you have a rank three tensor and `dims = {0, 1, 2}`. After the first two iterations, `sizes.size()` will be 1, which means that the last iteration `sizes.erase(sizes.begin() + 2 - 1)`, will try to remove an element out of bounds.

Also, I know this calculation was already here, but I don't think it is right. If `dimInt = 0`, then this line will try to erase a value out of bounds

Reply:
I can do one thing, undo the changes made to this function, and call it from the decomposition class directly in a loop for each dim to be reduced. Would that be fine?

Reply:
Yeah, since you're always calling it with `keepDim=True`, the approach you mentioned should be fine